File: AlbumLibrary/src/main/java/com/album/presenter/impl/PreviewPresenterImpl.java
package com.album.presenter.impl;

import android.content.ContentResolver;
import android.text.TextUtils;

import com.album.entity.AlbumEntity;
import com.album.entity.FinderEntity;
import com.album.presenter.PreviewPresenter;
import com.album.ui.view.PreviewView;
import com.album.ui.view.ScanView;
import com.album.ui.widget.ScanCallBack;
import com.album.util.scan.AlbumScanUtils;
import com.album.util.scan.VideoScanUtils;
import com.album.util.task.AlbumTask;
import com.album.util.task.AlbumTaskCallBack;

import java.util.ArrayList;
import java.util.List;

/**
 * by y on 17/08/2017.
 */

public class PreviewPresenterImpl implements PreviewPresenter, ScanCallBack {
    private final PreviewView previewView;
    private final ScanView scanView;

    public PreviewPresenterImpl(PreviewView previewView, boolean isVideo) {
        this.previewView = previewView;
        ContentResolver contentResolver = previewView.getPreviewActivity().getContentResolver();
        scanView = isVideo ? VideoScanUtils.get(contentResolver) : AlbumScanUtils.get(contentResolver);
    }

    @Override
    public void scan(final String bucketId, final int page, final int count) {
        previewView.getPreviewActivity().runOnUiThread(new Runnable() {
            @Override
            public void run() {
                previewView.showProgress();
            }
        });
        AlbumTask.get().start(new AlbumTaskCallBack.Call() {
            @Override
            public void start() {
                scanView.start(PreviewPresenterImpl.this, bucketId, page, count);
            }
        });
    }

    @Override
    public void mergeEntity(List<AlbumEntity> albumEntityList, ArrayList<AlbumEntity> selectAlbumEntityList) {
        if (albumEntityList == null || selectAlbumEntityList == null) {
            return;
        }
        for (AlbumEntity albumEntity : selectAlbumEntityList) {
            String path = albumEntity.getPath();
            for (AlbumEntity allAlbumEntity : albumEntityList) {
                String allEntityPath = allAlbumEntity.getPath();
                if (TextUtils.equals(path, allEntityPath)) {
                    allAlbumEntity.setCheck(albumEntity.isCheck());
                }
            }
        }
    }


    @Override
    public void scanSuccess(final ArrayList<AlbumEntity> albumEntityList, ArrayList<FinderEntity> list) {
        previewView.getPreviewActivity().runOnUiThread(new Runnable() {
            @Override
            public void run() {
                previewView.hideProgress();
                previewView.scanSuccess(albumEntityList);
            }
        });
    }

    @Override
    public void resultSuccess(AlbumEntity albumEntity, ArrayList<FinderEntity> finderEntityList) {

    }

}

File: AlbumLibrary/src/main/java/com/album/presenter/impl/PreviewPresenterImpl.kt
package com.album.presenter.impl

import android.text.TextUtils
import com.album.AlbumBundle
import com.album.AlbumEntity
import com.album.FinderEntity
import com.album.presenter.PreviewPresenter
import com.album.ui.view.PrevView
import com.album.util.AlbumTask
import com.album.util.AlbumTaskCallBack
import com.album.util.scan.AlbumScanUtils
import com.album.util.scan.ScanCallBack
import com.album.util.scan.ScanView
import java.util.*

/**
 * by y on 17/08/2017.
 */

class PreviewPresenterImpl(private val prevView: PrevView, private val albumBundle: AlbumBundle) : PreviewPresenter, ScanCallBack {

    private val scanView: ScanView = AlbumScanUtils[prevView.getPreViewActivity().contentResolver]

    override fun startScan(bucketId: String, page: Int, count: Int) {
        prevView.getPreViewActivity().runOnUiThread { prevView.showProgress() }
        AlbumTask.instance.start(object : AlbumTaskCallBack.Call {
            override fun start() {
                scanView.startScan(this@PreviewPresenterImpl, bucketId, page, count, albumBundle.filterImg, albumBundle.sdName)
            }
        })
    }

    override fun mergeEntity(albumEntityList: List<AlbumEntity>, selectEntity: ArrayList<AlbumEntity>) {
        for (albumEntity in selectEntity) {
            val path = albumEntity.path
            for (allAlbumEntity in albumEntityList) {
                val allEntityPath = allAlbumEntity.path
                if (TextUtils.equals(path, allEntityPath)) {
                    allAlbumEntity.isCheck = albumEntity.isCheck
                }
            }
        }
    }

    override fun scanCallBack(imageList: ArrayList<AlbumEntity>, finderList: ArrayList<FinderEntity>) {
        prevView.getPreViewActivity().runOnUiThread {
            prevView.hideProgress()
            prevView.scanSuccess(imageList)
        }
    }

    override fun resultCallBack(image: AlbumEntity?, finderList: ArrayList<FinderEntity>) {}

}
